src/bin/pg_autoctl/supervisor.c

```diff
@@ -845,6 +1378,24 @@ supervisor_update_pidfile(Supervisor *supervisor)
 }
 
 
+static bool
+supervisor_service_exists(Supervisor *supervisor, const char *serviceName)
+{
+	Supervisor_it sit;
+	Service *service;
+
+	foreach_supervised_service(&sit, supervisor, service)
+	{
+		if (!strncmp(service->name, serviceName, strlen(serviceName)))
```

Comment:
Please use `strncmp(...) == 0` or our macro `streq` if it's available in that module. The man page sayth that the return value is zero when the string is found. It sayth nothing about considering an integer as a boolean, and I don't trust C too much with that.